scripts/migrations.js

```diff
@@ -0,0 +1,79 @@
+import Constants from '../constants.js';
+import Settings from '../settings.js';
+
+export default class Migrations {
+	constructor() {
+		this._settings = new Settings();
+	}
+	/* Migrates the description of Custom Convenient Effects to use the ActiveEffect5e#description field instead of the legacy flags.
+	 *
+	 * @param {string} id - the ID of the custom effects item
+	 * @returns {Promise} a promise that resolves when all the Item's ActiveEffects descriptions have migrated
+	 */
+	async _migrateCustomEffectsItemEffectDescriptions(id, migration) {
+		const item = this._settings._findCustomEffectsItem(id);
+		const updates = [];
+		const effects = item.effects.filter((effect) =>
+			effect.getFlag(Constants.MODULE_ID, Constants.FLAGS.HAS_MIGRATED)
```

Comment:
I think I might have told you the wrong thing here. `effect.getFlag` can return undefined, which is falsey. If it returns undefined, it would filter out that effect. If it does not return undefined, it means it has been migrated and would return truthy. So... unless I'm totally confusing myself, you should do `!effect.getFlag`, right?

Reply:
Sorry for the confusion here. Please verify my logic is sound lol